### ISSUE TITLE
Units move faster diagonally

### ISSUE THREAD

**Issue by xezon** [xezon/GeneralsGamePatch#650]
Units move faster diagonally. Up to 1.41 times probably, as the diagonal of a rectangle is sqrt(1 + 1).

**Comment by Jundiyy** [comment]
Should this be treated as a bug? I don't know what kind of difference it would make, would weaker armies become even weaker as they now take even longer to travel across the map?

**Comment by xezon** [comment]
The math could be adjusted to that the new speed is always between the original min and max. Meaning if original min was 1 and max 1.41, then new speed could be 1.2. Then, no matter in which direction the unit moves, it always has the same speed.

**Comment by ImTimK** [comment]
I like consistency and it might be a good improvement. 

Though there might indeed be some balance impact.

As a player I utilize ’mechanic’ all the time and it helped me in a lot of situations, for example catching/trapping vees with my gats. 

Things like this wouldn’t be possible anymore.

**Comment by xezon** [comment]
We will not justify such bugs with balance in Game Patch. At the same token you could say that Humvee can escape diagonally faster from Gatling, making Humvee even better.

**Comment by Jundiyy** [comment]
Whilst having a 1.2 cap would make it equal in that sense, it would also mean all units now travel faster, this might make the game speed feel too fast to what players are used to. Of course it wouldn't take long to adjust although some things might do such as microing planes or other faster units.
Seems like a hard one imo.

**Comment by Jundiyy** [comment]
Actually I misunderstood what you said a bit, this would be even more tricky as the diagonal speed would then be reduced. And same for vice versa. Even more trickier.

**Comment by ImTimK** [comment]
> 

Yes, this might be one of the reasons vees are so strong in the open. However when they don't have much room to maneuver they can be caught with slower units.

The nerf of the first case might have more impact than the second case, so that would be positive. But it’s hard atm to predict the overall impact. 

Another positive case would be Lix survival, on many maps they can only flee vertically or horizontally along the edge of the map, units like Quads can currently intercept them quite often by cutting off their path diagonally. Which is good thing for China.

**Comment by MTKing4** [comment]
In favor of fixing the bug, the vee gat example doesn't make sense because it's a double edged sword that is affecting the interaction both ways

a lot of the times vees escaped because they were moving diagonally, and it's easier for them to do so because they have tires not treads and not needing to rotate to the intended direction before moving

**Comment by RisingZH** [comment]
We already discussed this one here:

https://github.com/xezon/GeneralsGamePatch/issues/123

**Comment by penfriendz** [comment]
I think this bug should be fixed. It doesn't make sense and in some ways impairs better game design choices when it comes to making decisions about specific units. Fixing the bug but then increasing the speed of all units by sqrt(1.5) seems like a good solution.

**Comment by commy2** [comment]
> but then increasing the speed of all units by sqrt(1.5)

(1+sqrt(2))/2
for the average :)

**Comment by xezon** [comment]
This bug will be fixed with Thyme when so possible.
